Show codepoint in hex with description and char.

tmalsburg opened the pull request:
Addresses #889. I'm not sure if `helm-ucs-max-len` is the max length of the description or the max length of the line.  If it's the length of the description, the PR should be ready to be merged.  If not, I have to amend it.  Adding a docstring to `helm-ucs-max-len` is a good idea either way.

thierryvolpiatto commented:
Hi Titus,

Titus von der Malsburg <notifications@github.com> writes:

> I'm not sure if helm-ucs-max-len is the max length of the description
> or the max length of the line. If it's the length of the description,
> the PR should be ready to be merged. If not, I have to amend
> it. Adding a docstring to helm-ucs-max-len is a good idea either way.
Yes it is the length of the car of each elements, it is used to align all.

Ok, looks good, but insertion is wrong, you have to modify the action:

(defun helm-ucs-insert-char (candidate)
  (with-helm-current-buffer
    (insert
     (replace-regexp-in-string
      " " ""
      (car (split-string
            (cadr (split-string candidate ":"))))))))

And probably you want an action for inserting the hex code:

(defun helm-ucs-insert-hex (candidate)
  (with-helm-current-buffer
    (insert
     (replace-regexp-in-string
      " " ""
      (cadr (split-string
            (cadr (split-string candidate ":"))))))))

And finally add it to the source (If you have time you can rewrite it
with helm-build-in-buffer-source).

Many thanks ;-)


-- 
Thierry
Get my Gnupg key:
gpg --keyserver pgp.mit.edu --recv-keys 59F29997

tmalsburg commented:
@thierryvolpiatto, I rewrote the code for inserting characters.  The previous version failed when then inserted character was the space character.  I also added insert functions for the name and the code point of a character.

Why is `with-helm-current-buffer` needed here? Inserting seems to work just as well without it.

thierryvolpiatto commented:
Thanks.

> Why is with-helm-current-buffer needed here? Inserting seems to work just as well without it.
It may work most of the time but may fail in some corner cases, using `with-helm-current-buffer` ensure insertion in the right place.
